src/graph/FetchVerticesExecutor.cpp

```diff
@@ -155,277 +350,293 @@ void FetchVerticesExecutor::fetchVertices() {
-    } else {
-        status = setupVidsFromExpr();
     }
 
-    return status;
-}
-
-Status FetchVerticesExecutor::setupVidsFromExpr() {
-    Status status = Status::OK();
-    std::unique_ptr<std::unordered_set<VertexID>> uniqID;
-    if (distinct_) {
-        uniqID = std::make_unique<std::unordered_set<VertexID>>();
+    if (yieldClause_ == nullptr) {
+        for (TagID tagId : tagIdSet) {
+            auto tagSchema = tagSchemaMap[tagId];
+            auto tagFound = ectx()->schemaManager()->toTagName(spaceId_, tagId);
+            if (!tagFound.ok()) {
+                VLOG(3) << "Tag name not found for tag id: " << tagId;
+                // Ignore the bad data.
+                continue;
+            }
+            auto tagName = std::move(tagFound).value();
+
+            for (auto iter = tagSchema->begin(); iter != tagSchema->end(); ++iter) {
+                auto *ref = new std::string("");
+                auto *alias = new std::string(tagName);
+                auto *prop = iter->getName();
+                Expression *expr =
+                    new AliasPropertyExpression(ref, alias, new std::string(prop));
+                auto *column = new YieldColumn(expr);
+                yieldColsHolder_.addColumn(column);
+                yields_.emplace_back(column);
+                colNames_.emplace_back(expr->toString());
+                colTypes_.emplace_back(nebula::cpp2::SupportedType::UNKNOWN);
+            }
+        }
     }
 
-    expCtx_->setSpace(spaceId_);
-    auto vidList = sentence_->vidList();
-    Getters getters;
-    for (auto *expr : vidList) {
-        expr->setContext(expCtx_.get());
-        status = expr->prepare();
-        if (!status.ok()) {
-            break;
-        }
-        auto value = expr->eval(getters);
-        if (!value.ok()) {
-            return value.status();
-        }
-        auto v = value.value();
-        if (!Expression::isInt(v)) {
-            status = Status::Error("Vertex ID should be of type integer");
-            break;
+    if (fromType_ == kRef) {
+        if (inputs_p_ == nullptr) {
+            LOG(ERROR) << "inputs is nullptr.";
+            doError(Status::Error("inputs is nullptr."));
+            return;
         }
 
-        auto valInt = Expression::asInt(v);
-        if (distinct_) {
-            auto result = uniqID->emplace(valInt);
-            if (result.second) {
-                vids_.emplace_back(valInt);
+        auto visitor = [&, this] (const RowReader *reader) -> Status {
+            VertexID vid = 0;
+            auto rc = reader->getVid(*colname_, vid);
+            if (rc != ResultType::SUCCEEDED) {
+                return Status::Error("Column `%s' not found", colname_->c_str());
             }
-        } else {
-            vids_.emplace_back(valInt);
-        }
-    }
+            if (dataMap.find(vid) == dataMap.end() && !expCtx_->hasInputProp()) {
+                return Status::OK();
+            }
+            auto& ds = dataMap[vid];
```

Comment:
You could use the iterator returned by find directly.

Reply:
yes